v1/format/format.go

```diff
@@ -1570,14 +1803,16 @@ func (w *writer) endLine() {
 }
 
 // beforeLineEnd registers a comment to be printed at the end of the current line.
-func (w *writer) beforeLineEnd(c *ast.Comment) {
+func (w *writer) beforeLineEnd(c *ast.Comment) error {
 	if w.beforeEnd != nil {
 		if c == nil {
-			return
+			return nil
 		}
-		panic("overwriting non-nil beforeEnd")
+		w.beforeEnd = nil
+		return errors.New("unexpected comment")
```

Comment:
The original error message won't win any rewards in clarity 😄, but is this much more clear? 
Can we improve this somehow? The actual issue is that we're lining up multiple comments to be printed at the end of the current line?

Reply:
How about `fmt.Errorf("unexpected new comment because there is already a comment registered for the current line %d", c.Location.Row)` ? Then maybe the user can get past this issue by fixing the comment at the printed location? Could also print the raw comment text to help find it. 

Hopefully the user never sees this error msg but of course there could be edge cases missed that it doesn't revert it correctly.

Reply:
Truncating with an ellipsis is a good idea. I don't see the line number in the rego file printed, just the error message and the stack trace. maybe I am missing it? How about:

`unexpected new comment (# test 1) on line 9 because there is already a comment (# test 2) registered for line 9`